docs/js/immutablejs.md

```diff
@@ -74,3 +74,33 @@ const initialState = new StateRecord({}); // initialState is now a new StateReco
 
 Now, if you want to access `myData`, you can just write `state.myData` in your reducer code.
 
+## Making Complex Mutations Efficiently
+
+Because a new instance is created and returned each time a value is set or updated on an immutable data structure, complex mutations that aim to make changes to multiple parts of the data structure can cause a performance penalty.  
+To avoid creating multiple instances of the data structure, we can utilise ImmutableJS's [`.withMutations`](https://facebook.github.io/immutable-js/docs/#/Map/withMutations), batching all of our changes in to a single mutation.
+
+```JS
+const stateMap = Map({
+    a: 'A',
+    b: 'B',
+    c: 'C'
+});
+
+// Creates 3 new Map instances
+const newState = stateMap
+					.set('a', 1)
```

Comment:
nit: please use 2 spaces to match project style guidelines

code style looks mostly good, i would just quickly paste these blocks into project file for linting just be to sure